dist/mushroom-strategy.js

```diff
@@ -1,1335 +1,1210 @@
+            switches,
+            entityConfig,
+            {
+              type: "custom:mushroom-entity-card",
+              tap_action: {
+                action: "toggle",
+              },
+            },
+            this.#createTitleCard(
+                null,
+                "Switches",
+                "switch.turn_off",
+                "switch.turn_off",
+                "mdi:power-plug-off",
+                "mdi:power-plug",
+                area.area_id,
+            ),
+        ),
+      });
+    }
+
+    // Create a column of climate cards.
+    const climates = this.#getDeviceEntities(area, "climate");
+
+    if (climates.length) {
+      cards.push({
+        type: "vertical-stack",
+        cards: this.#createViewCards(
+            climates,
+            entityConfig,
+            {
+              type: "custom:mushroom-climate-card",
+              hvac_modes: [
+                "off",
+                "cool",
+                "heat",
+                "fan_only",
+              ],
+              show_temperature_control: true,
+            },
+            {
+              type: "custom:mushroom-title-card",
+              subtitle: "Climate",
+            },
+        ),
+      });
+    }
+
+    // Create a column of Media player cards.
+    const mediaPlayers = this.#getDeviceEntities(area, "media_player");
+
+    if (mediaPlayers.length) {
+      cards.push({
+        type: "vertical-stack",
+        cards: this.#createViewCards(
+            mediaPlayers,
+            entityConfig,
+            {
+              type: "custom:mushroom-media-player-card",
+              use_media_info: true,
+              media_controls: [
+                "on_off",
+                "play_pause_stop",
+              ],
+              show_volume_level: true,
+              volume_controls: [
+                "volume_mute",
+                "volume_set",
+                "volume_buttons",
+              ],
+            },
+            {
+              type: "custom:mushroom-title-card",
+              subtitle: "Media Players",
+            },
+        ),
+      });
+    }
+
+    // Create a column of Sensor cards.
+    let sensors        = this.#getDeviceEntities(area, "sensor");
+    const sensorStates = this.#getStateEntities(area, "sensor");
+    const sensorCards  = [];
+
+    if (sensors.length) {
+      // Add a Title card.
+      sensorCards.push({
+        type: "custom:mushroom-title-card",
+        subtitle: "Sensors",
+      });
+
+      // Create a card for each sensor.
+      for (const sensor of sensors) {
+        // Find the state of the current sensor.
+        const sensorState = sensorStates.find(state => state.entity_id === sensor.entity_id);
+
+        // Define the card type.
+        let card = {
+          type: "custom:mushroom-entity-card",
+          entity: sensor.entity_id,
+          icon_color: "green",
+        };
+
+        if (sensorState.attributes.unit_of_measurement) {
```

Comment:
Not every sensor state has attributes.
.... For some reason.
Getting error Following error for one of my areas
```
TypeError: Cannot read properties of undefined (reading 'attributes')
```

Reply:
This means `sensorState` is `undefined` which is strange because...

`sensorState` is the entity of `sensorStates` with the same `entity_id`.
`sensorStates` contains entities of `info.hass.states` with the same domain and area as your sensor (or no area at all).
`info.hass.states` contains all entities of Home Assistant with a state.

All three contain entities, with a property `entity_id` which should link them all together.

Can you replicate the issue with a sensor I could have?
or share the values of `sensor` and `info.hass.states["whatever the troubled sensor's entity_id is"]`?

Reply:
Yeah this seems a bit of an odd issue. I found out the entity causing the trouble is a sensor from the [Node red companion integration](https://github.com/zachowj/hass-node-red). `sensor.islamic_month`

Reply:
Is it perhaps a RESTful Sensor, configured in your yaml?
If so, I would like to see your yaml.
Otherwise I need you to share the values of `sensor` and `info.hass.states["sensor.islamic_month"]`?

Reply:
Sorry about the late reply
Yes you are correct the condition should be 'any'
The example code is given in the custom strategies home assistant docs also say any